### PR TITLE
add maintenance workflow

### PR DESCRIPTION
**Pull request by bdemann**
- [x] Breaking changes enumerated in [the release issue](https://github.com/demergent-labs/azle/issues/2053)
- [x] New documentation enumerated in [the release issue](https://github.com/demergent-labs/azle/issues/2053)
- [x] Code has been declaratized
- [x] Error handling beautiful (no unwraps or expects etc)
- [x] Code tested thoroughly
- [x] All new functions have JSDoc/Rustdoc comments

### CODE REVIEW

File: .github/workflows/maintenance.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+name: Maintenance
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:1]: Call it `Examples/tests maintenance (npm install, prettier, eslint, etc)`

File: .github/workflows/maintenance.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            - uses: actions/checkout@v4
+              with:
+                  ref: ${{ needs.create-branch-prefix.outputs.base-branch }}
+                  token: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
+
+            - uses: ./.github/actions/setup_node
+
+            - run: npm install
+
+            # Run global prettier/lint fixes
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:37]: Use our new fancy scripts to do the pretty and lint stuff

File: .github/workflows/maintenance.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            - name: Run Prettier
+              run: npx prettier --write .
+
+            - name: Run ESLint
+              run: npx eslint . --fix
+
+            - uses: ./.github/actions/commit_and_push
+              with:
+                  branch-name: ${{ needs.create-branch-prefix.outputs.base-branch }}
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:46]: Should this output be specified in the create-branch-prefix step?

File: .github/workflows/maintenance.yml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+                  commit-message: 'Run global prettier and eslint fixes'
+                  gpg-signing-key: ${{ secrets.GPG_SIGNING_KEY }}
+                  signing-key-id: ${{ inputs.signing-key-id }}
+
+            - id: get-test-infos
+              uses: ./.github/actions/get_test_infos
+              with:
+                  directories: |
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:54]: These are wrong

File: .github/workflows/maintenance.yml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            - name: Run Prettier
+              run: npx prettier --write .
+
+            - name: Run ESLint
+              run: npx eslint . --fix
+
+            - uses: ./.github/actions/commit_and_push
+              with:
+                  branch-name: ${{ needs.create-branch-prefix.outputs.base-branch }}
+                  commit-message: 'Run global prettier and eslint fixes'
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:47]: Change the commit message to the new name of the workflow

File: .github/workflows/maintenance.yml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+              uses: ./.github/actions/get_test_infos
+              with:
+                  directories: |
+                      ./examples
+                      ./tests
+
+    update-test-dependencies:
+        needs:
+            - prepare-maintenance
+            - create-branch-prefix
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:61]: Remove dependency on create-branch-prefix, do for all

File: .github/workflows/maintenance.yml [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+on:
+    workflow_dispatch:
+        inputs:
+            signing-key-id:
+                required: true
+                type: string
+                default: C8B77BCBE16CD2B94B43F9C8757397B82D4ED7B0
+
+jobs:
+    create-branch-prefix:
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:11]: Shouldn't there be outputs here?

File: .github/workflows/maintenance.yml [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+              working-directory: ${{ matrix.test.path }}
+              run: |
+                  rm -f package-lock.json
+                  npm install
+
+            - name: Create branch name
+              id: create-branch-name
+              uses: ./.github/actions/create_branch_name
+              with:
+                  prefix: ${{ needs.create-branch-prefix.outputs.branch-prefix }}
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:90]: Is this output up there?

File: .github/workflows/maintenance.yml [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+                  ref: ${{ github.ref }}
+
+            - name: Create Pull Request
+              env:
+                  GH_TOKEN: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
+              run: |
+                  gh pr create \
+                    --base main \
+                    --head ${{ needs.create-branch-prefix.outputs.base-branch }} \
+                    --title "Maintenance: Update dependencies and fix formatting" \
```
Comment (lastmjs) [inline .github/workflows/maintenance.yml:135]: Change all title, names, commit messages to the same thingish

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: .github/workflows/maintenance.yml (hunk 1 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:1]: done

File: .github/workflows/maintenance.yml (hunk 7 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:11]: Apparently when you do a reusable workflow like this you CANNOT specify the outputs except in the workflow yml.

File: .github/workflows/maintenance.yml (hunk 2 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:37]: I changed my mind, these are great and maybe even preferable

File: .github/workflows/maintenance.yml (hunk 3 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:46]: apparently not

File: .github/workflows/maintenance.yml (hunk 4 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:54]: done

File: .github/workflows/maintenance.yml (hunk 8 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:90]: It shouldn't be

File: .github/workflows/maintenance.yml (hunk 9 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:135]: Done

File: .github/workflows/maintenance.yml (hunk 6 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:61]: Basically there are two reasons to specify a dependency. One is to establish the run order of the jobs, the other is to get values from outputs. If you only care about the first then you only need the one dependency that run directly before it, but to get the second you need to specify all the dependencies you want the outputs from. I simplified all the ones I could, and left the rest

File: .github/workflows/maintenance.yml (hunk 5 above)
Comment (bdemann) [reply .github/workflows/maintenance.yml:47]: done \

**Review by lastmjs — APPROVED** [review]

**Pull request merged**